lib/std/atomic.zig

```diff
@@ -3,14 +3,88 @@
 // This file is part of [zig](https://ziglang.org/), which is MIT licensed.
 // The MIT license requires this copyright notice to be included in all copies
 // and substantial portions of the software.
+
+const std = @import("std.zig");
+const target = std.Target.current;
+
+pub const Ordering = std.builtin.AtomicOrder;
+
 pub const Stack = @import("atomic/stack.zig").Stack;
 pub const Queue = @import("atomic/queue.zig").Queue;
-pub const Bool = @import("atomic/bool.zig").Bool;
-pub const Int = @import("atomic/int.zig").Int;
+pub const Atomic = @import("atomic/Atomic.zig").Atomic;
 
 test "std.atomic" {
     _ = @import("atomic/stack.zig");
     _ = @import("atomic/queue.zig");
-    _ = @import("atomic/bool.zig");
-    _ = @import("atomic/int.zig");
+    _ = @import("atomic/Atomic.zig");
+}
+
+pub fn fence(comptime ordering: Ordering) callconv(.Inline) void {
+    switch (ordering) {
+        .Acquire, .Release, .AcqRel, .SeqCst => {
+            @fence(ordering);
+        },
+        else => {
+            @compileLog(ordering, " only applies to a given memory location");
+        },
+    }
+}
+
+pub fn compilerFence(comptime ordering: Ordering) callconv(.Inline) void {
+    switch (ordering) {
+        .SeqCst => {
+            // TODO: Use LLVM intrinsic `fence syncscope("singlethread")` instead of inline asm.
```

Comment:
There are some [unresolved problems](https://github.com/rust-lang/rust/issues/62256) with the use of singlethread fences, the biggest one is that they may produce extra instructions or a call (!) to `__sync_synchronize`.
Let's use the good old empty-asm-block trick for the time being (and group the ordering parameters, having to expand 3 layers of recursion is pretty bad for stage1, beside being hard to understand).

Reply:
Weird find.. 

fixed it